checker/src/visitors.rs

```diff
@@ -1864,12 +1882,27 @@ impl<'a, 'b: 'a, 'tcx: 'b, E> MirVisitor<'a, 'b, 'tcx, E> {
                     ..
                 } = constant.borrow();
                 let const_value = self.visit_constant(ty, *user_ty, &literal.val);
-                if let Expression::AbstractHeapAddress(ordinal) = const_value.expression {
-                    let rtype = ExpressionType::Reference;
-                    let rpath = Rc::new(PathEnum::AbstractHeapAddress { ordinal }.into());
-                    self.copy_or_move_elements(path, rpath, rtype, false);
-                } else {
-                    self.current_environment.update_value_at(path, const_value);
+                match &const_value.expression {
+                    Expression::AbstractHeapAddress(ordinal) => {
+                        let rtype = ExpressionType::Reference;
+                        let rpath =
+                            Rc::new(PathEnum::AbstractHeapAddress { ordinal: *ordinal }.into());
+                        self.copy_or_move_elements(path, rpath, rtype, false);
+                    }
+                    Expression::CompileTimeConstant(val) => {
```

Comment:
It only really makes sense to do this for constants that have structure which is visible to user code, which is to say string constants.